app/web/src/features/messages/requests/MarkAllReadButton.tsx

```diff
@@ -0,0 +1,66 @@
+import { Tooltip } from "@material-ui/core";
+import IconButton from "components/IconButton";
+import { DoneAllIcon } from "components/Icons";
+import { MARK_ALL_READ } from "features/messages/constants";
+import { useMutation } from "react-query";
+import { service } from "service";
+import getAllPages from "utils/getAllPages";
+
+export default function MarkAllReadButton({
+  type,
+}: {
+  type: "chats" | "hosting" | "surfing";
+}) {
+  const markAll = useMutation(async () => {
```

Comment:
I'm guessing this could probably be simplified once host requests are separated from chat messages as it shouldn't affect the notification badge?